### ISSUE TITLE
[Document list] List items are not removed correctly with backspace

### ISSUE THREAD

**Issue by FilipTokarski** [ckeditor/ckeditor5#11347]
Part of #11377.

## 📝 Provide detailed reproduction steps (if any)

1.  Switch to `ck/10812-document-list-editing`
2.  Create a few list items
3.  In an empty list item press `backspace`

### ✔️ Expected result

Item gets removed, cursor goes up to the previous list item.

### ❌ Actual result

Cursor does not go to the previous list item. All other list items get spaced out, which looks bad. Only after second `backspace` they get back to normal.

### Screencast

https://user-images.githubusercontent.com/34380544/155957333-e7e6f6f8-15a3-40ad-ab3b-51ca38b44eaa.mp4

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Reinmar** [comment]
That's how it works on Google Docs:

https://user-images.githubusercontent.com/156149/156347235-d3385a63-03f8-4046-a56d-25d3bc179ed9.mp4

Note, that I have paragraphs ("Normal text") style in Google Docs to have bigger space after them:

<img width="1038" alt="image" src="https://user-images.githubusercontent.com/156149/156347401-21b42592-a3fd-4e9f-87f1-f9870df879a8.png">

If you'd test it with the default styles that Google Docs come with (no spacing around paragraphs), it would look a bit differently:


https://user-images.githubusercontent.com/156149/156347822-bc8d6074-7375-4517-b38f-c5ad56c99f82.mp4

To sum up:

* The document changes as expected.
* We could change the styling a bit to make it less confusing.

**Comment by Reinmar** [comment]
Goals:

*   Try to limit the visual noise when merging two paragraphs into one list item
*   But don't overdo it – let's not overcomplicate the stylesheets

**Comment by oleq** [comment]
```css
.ck-content li > p {
    margin: 0;
}
```

for starters and let's see where do we go from here.

**Comment by Reinmar** [comment]
Related ticket: #16866.

**Comment by Reinmar** [comment]
Related (not functionally, but due to combined confusion that these issues create together): #16884.
